events/api.py

```diff
@@ -1847,11 +1847,43 @@ class RegistrationSerializer(LinkedEventsSerializer, RegistrationBaseSerializer)
         queryset=Event.objects.all(),
     )
 
-    only_admin_visible_fields = (
-        "created_by",
-        "last_modified_by",
-        "registration_user_accesses",
-    )
+    # LinkedEventsSerializer removes created_by and last_modified_by default, so set this
+    # to empty array and have custom logic to hide admin fields in to_representation
+    only_admin_visible_fields = []
```

Comment:
This is a bit confusing, so instead of doing this and overriding `to_representation`, I'd rather refactor `LinkedEventsSerializer` a bit.

There's this bit of code in `LinkedEventsSerializer.to_representation`:
```py
        # display non-public fields if 1) obj has publisher org and 2) user belongs to the same org tree
        # never modify self.skip_fields, as it survives multiple calls in the serializer across objects
        obj_skip_fields = set(self.skip_fields)
        if (
            self.user
            and hasattr(obj, "publisher")
            and obj.publisher
            and obj.publisher.tree_id in self.admin_tree_ids
        ):
            for field in self.only_admin_visible_fields:
                obj_skip_fields.remove(field)
        for field in obj_skip_fields:
            if field in ret:
                del ret[field]
        return ret
```
...and this bit in its `__init__`:
```py
        # by default, admin fields are skipped
        self.skip_fields = skip_fields | set(self.only_admin_visible_fields)
```

The only code that refers to `only_admin_visible_fields` is `LinkedEventsSerializer`, and, of course, this serializer, while `skip_fields` is only referred in serializer contexts, so they should be safe to modify without causing any behavioral oddity.

What I'd change is:
1. `self.skip_fields` doesn't automatically append `self.only_admin_visible_fields` into itself.
2. Instead of removing `only_admin_visible_fields` from `obj_skip_fields` if the user is an admin, do the inverse (so add the fields if the user is not an admin)
3. Do some code splitting; split the admin check part and/or the part where you get the fields to skip to its own method(s)
4. In `RegistrationSerializer`, override the new method(s) to your liking, e.g. override the admin check part to whatever's written in `to_representation` at the moment
5. In `RegistrationSerializer`, use `only_admin_visible_fields` as usual.

This should make the code a bit more logical and less complex.

Reply:
Very nice suggestion. 
I added `are_only_admin_visible_fields_allowed` method to check are admin visible fields allowed. `only_admin_visible_fields` are appended to skip_fields only if the user is admin.
I override that method in `RegistrationSerializer` to allow also registration admin users to see admin fields

Changed are committed in https://github.com/City-of-Helsinki/linkedevents/pull/715/commits/2f71eb4838d2c710885ae548e5705d1d69817c2c

Reply:
Well done!